README.md

```diff
@@ -1,4 +1,1060 @@
+
+For the caches to work properly, it is sufficient that the following two lines are included in your `/etc/nix/nix.conf`:
+```txt
+trusted-users = USER
+experimental-features = nix-command flakes
+```
+Replace `USER` with the result of running `whoami`. 
+
+You may need to reload the nix daemon on Darwin for changes to `/etc/nix/nix.conf` to take effect:
+```bash
+sudo launchctl stop org.nixos.nix-daemon
+sudo launchctl start org.nixos.nix-daemon
+```
+Leave `allow-import-from-derivation` set to `true` for `haskell.nix` for work correctly.
+
+## 3.2. `nix/iogx-config.nix`
+
+```nix
+# Example of a valid nix/iogx-config.nix
+{ 
+  repoRoot = ../.; 
+  systems = [ "x86_64-linux" "x86_64-darwin" ]; 
+  haskellCompilers = [ "ghc8107" ]; 
+  defaultHaskellCompiler = "ghc8107"; 
+  shouldCrossCompile = true; 
+} 
+```
+
+This file must exist and it contains the main configuration values for your project.
+
+### 3.2.1. `repoRoot` 
+
+A Nix path to an existing folder containing your `cabal.project` file.
+
+In future versions this field will be removed and it will default to the repository top-level directory, but until then it must be explicitly set (usually to `../.`).
+
+### 3.2.2. `systems`
+
+The non-empty list of systems against which your project can be built. 
+
+These are standard Nix values found in `nixpkgs.stdenv.system`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.3. `haskellCompilers`
+
+The non-empty list of GHC versions that can build your project. 
+
+Currently two GHC versions are supported and provided by IOGX: `ghc8107` and `ghc927`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.4. `defaultHaskellCompiler`
+
+Only one compiler at a time can be visible in the `$PATH` in the shell.
+
+When calling `nix develop`, the `defaultHaskellCompiler` will be selected. 
+
+To enter a shell with a different compiler, see to the [flake outputs](#310-flake-outputs-format) section. 
+
+This field is optional and defaults to the first (leftmost)  compiler in [`haskellCompilers`](#323-haskellcompilers).
+
+### 3.2.5. `shouldCrossCompile`
+
+Cross-compilation on Windows is available via `mingwW64` on `x86_64-linux` only. 
+
+If your project cannot be cross-compiled then set this field to `false`. 
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is optional and defaults to `true`.
+
+## 3.3. `nix/haskell-project.nix`
+
+```nix
+# Default nix/haskell-project.nix
+{ inputs 
+, inputs' 
+, pkgs 
+, { haskellCompiler
+  , enableHaddock 
+  , enableProfiling 
+  , enableCross 
+  }@meta 
+}:
+{
+  cabalProjectLocal = ""; 
+  sha256map = { }; 
+  shellWithHoogle = false; 
+  modules = [ ]; 
+  overlays = [ ];
+}
+```
+
+This file describes your Haskell project and will be used internally to call `haskell.nix:cabalProject'`.
```

Comment:
Instead of saying how it's used internally, should probably just link to the haskell.nix docs for the arguments.

Reply:
And perhaps say "some args will be defaulted based on your repo layout and common heuristics, but can be overridden here" (assuming they can be overridden).

Reply:
Or is this only a subset of `h.nix` args? If so I'd say something like that.